photon-api/src/main/scala/com/linkedin/photon/ml/estimators/GameEstimator.scala

```diff
@@ -192,6 +197,7 @@ class GameEstimator(val sc: SparkContext, implicit val logger: Logger) extends P
     setDefault(computeVariance, false)
     setDefault(treeAggregateDepth, DEFAULT_TREE_AGGREGATE_DEPTH)
     setDefault(ignoreThresholdForNewModels, false)
+    setDefault(featureShardStats, None:FeatureShardStatisticsMapOpt)
   }
 
   /**
```

Comment:
i am more inlined to op2 for flexibility.

Reply:
Just spoke with @YazhiGao offline to clarify the options:

If `A = set of FeatureShardId for summaries` and `B = set of FeatureShardId for RE configurations`, then we have 4 options:

1. A = B
2. A ⊂ B
3. B ⊂ A
4. None of the above

The implications of these choices are as follows:

1. The feature summaries must be pruned of non-random-effect shard summaries and feature filtering cannot be disabled selectively.
2. The feature summaries must be pruned of non-random-effect shard summaries.
3. Feature filtering cannot be disabled selectively.
4. No requirements: if a random effect has a summary, feature filtering will be enabled.

I don't like the idea of making the user prune the feature summaries, so that rules out options 1 and 2. Between options 3 and 4: I don't see why a user would want to disable feature filtering if it is useful, however I don't think we should force them to either. Thus I personally favour option 4.